Doberman/BaseDevice.py

```diff
@@ -207,16 +207,23 @@ def setup(self):
             # so we can more easily change this later
             self.msg_sleep = 1.0
 
-        if self.tty == '0':
+        if hasattr(self, 'id'):
+            self._device.port = f'/dev/serial/by-id/{self.id}'
+            try:
+                self._device.open()
+            except serial.SerialException as e:
+                raise ValueError(f'Problem opening {self._device.port}: {e}')
+        elif self.tty == '0':
             raise ValueError('No tty port specified!')
-        try:
-            if self.tty.startswith('/'):  # Full path to device TTY specified
-                self._device.port = self.tty
-            else:
-                self._device.port = f'/dev/tty{self.tty}'
-            self._device.open()
-        except serial.SerialException as e:
-            raise ValueError(f'Problem opening {self._device.port}: {e}')
+        else:
+            try:
```

Comment:
Can you move the try outside the if to combine both exception handling parts (since they are the same)